standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnStore.java

```diff
@@ -461,9 +480,13 @@ void onRename(String oldCatName, String oldDbName, String oldTabName, String old
    * @return list of CompactionInfo structs.  These will not have id, type,
    * or runAs set since these are only potential compactions not actual ones.
    */
+  @SqlRetry
+  @Transactional(POOL_COMPACTOR)
```

Comment:
should we set `@Transactional(POOL_COMPACTOR)` in CompactionTxnHandler?

Reply:
The Method argument in ProxyTxnHandler points to the TxnStore method, there is no way to access the annotation if put on the implementing class.